src/arguments/song.ts

```diff
@@ -73,4 +83,42 @@ export default class extends Argument {
 			.filter((track) => (allowStreams ? true : track.info.isStream) && track.info.length <= maximumDuration)
 			.slice(0, remainingUserEntries);
 	}
+
+	private async loadQueue(message: KlasaMessage, url: string) {
+		try {
+			const rawData = await fetch<string[]>(url, FetchResultTypes.JSON);
+			return Promise.all(
+				rawData.map(async (track) => {
+					return {
+						track,
+						info: await this.decodeTrack(track)
+					};
+				})
+			);
+		} catch {
+			throw message.language.get('musicManagerImportQueueError');
+		}
+	}
+
+	private decodeTrack(track: string) {
+		const node = this.client.lavalink.idealNodes[0];
+		const params = new URLSearchParams();
+		params.append('track', track);
+		return fetch<DecodeResponse>(
+			`http://${node.host}:${node.port}/decodetrack?${params}`,
```

Comment:
Any particular reason why you're not using `Rest.decode` here?

Reply:
Rest.decode's types are fucked